### PR TITLE
Refactoring the use of temporary files in tpm_main.py

### PR DESCRIPTION
**Pull request by ShubhikaBhardwaj**
Please fill in the fields below to submit a pull request. The more information
that is provided, the better.

**Fixes issue #1067  **:

Description of the changes being introduced by the pull request:

For Code improvement, refactoring the use of temporary files in tpm_main.py 
- Refactored methods such as _tpm2_checkquote and _tpm2_printquote in tpm class to follow PEP 8 Python standards
- Documenting these functions using pydocs

### CODE REVIEW

File: keylime/tpm/tpm_main.py [hunk 1]
```diff
@@ -1039,51 +1039,77 @@ def create_quote(
...
+                pcr_file,
                 "-g",
                 hash_alg,
                 "-q",
                 nonce,
             ]
 
-        retDict = self.__run(command, lock=False)
-        return retDict
+        ret_dict = cmd_exec.run(command, lock=False)
```
Comment (aplanas) [inline keylime/tpm/tpm_main.py:1098]: that change it is not the same, the local `__run` should be indeed rewritten, but you are using parameters that make sense in `__run` but not in `run`, like `lock`

the rename of variables makes sense

File: keylime/tpm/tpm_main.py [hunk 2]
```diff
@@ -1039,51 +1039,77 @@ def create_quote(
...
+        """Executes the tpm2_print command.
+
+        Args:
+            quote_file: The quote file to print.
+
+        Returns:
+            A dictionary containing the stdout and stderr of the command.
+        """
+        command = ["tpm2_print", "-t", "TPMS_ATTEST", quote_file]
+        ret_dict = cmd_exec.run(command, lock=False)
```
Comment (aplanas) [inline keylime/tpm/tpm_main.py:1111]: same

**Comment by aplanas** [comment]
> **Fixes issue https://github.com/keylime/keylime/issues/1067 **

It is still not addressing the pointed issue?

**Comment by aplanas** [comment]
Also: wellcome : )

File: keylime/tpm/tpm_main.py (hunk 1 above)
Comment (stefanberger) [reply keylime/tpm/tpm_main.py:1098]: > the rename of variables makes sense

It only makes sense if we all agree that we don't want CamelCase but underscore everywhere from now on.

Comment (aplanas) [reply keylime/tpm/tpm_main.py:1098]: The agreement is mostly implicit, as we are following pep8:

https://peps.python.org/pep-0008/#function-and-variable-names

```
mixedCase is allowed only in contexts where that’s already the prevailing style (e.g. threading.py), to retain backwards compatibility.
```

File: keylime/tpm/tpm_main.py [hunk 3]
```diff
@@ -1039,51 +1039,77 @@ def create_quote(
...
-        retDict = self.__run(command, lock=False)
-        return retDict
+        ret_dict = cmd_exec.run(command, lock=False)
+        return ret_dict
 
-    def __tpm2_printquote(self, quoteFile: str) -> cmd_exec.RetDictType:
-        command = ["tpm2_print", "-t", "TPMS_ATTEST", quoteFile]
-        retDict = self.__run(command, lock=False)
-        return retDict
+    def tpm2_printquote(self, quote_file: str) -> cmd_exec.RetDictType:
```
Comment (stefanberger) [inline keylime/tpm/tpm_main.py:1101]: You would have to adjust the callers as well.

File: keylime/tpm/tpm_main.py [hunk 4]
```diff
@@ -1039,51 +1039,77 @@ def create_quote(
 
         return "r" + quote
 
-    def __tpm2_checkquote(
-        self, pubaik: str, nonce: str, quoteFile: str, sigFile: str, pcrFile: str, hash_alg: Union[Hash, str]
+    def tpm2_checkquote(
```
Comment (stefanberger) [inline keylime/tpm/tpm_main.py:1042]: You changed the function from a private to a public function. You would at least have to adjust all callers as well.

**Comment by stefanberger** [comment]
Is this PR still alive? The title states 'Refactoring the use of temporary files in tpm_main.py' but it doesn't seem to address this particular issue at all...

**Comment by ShubhikaBhardwaj** [comment]
Hi @stefanberger  
I am working on your suggestions.

**Comment by stefanberger** [comment]
I think this PR and its related issue are obsoleted by the fact that we are getting rid of usage of tpm2-tools and temporary files in tpm_main.py won't be necessary anymore.

**Comment by stefanberger** [comment]
Can we close this PR? At least all the code section that were supposed to be change have been moved out of tpm_main.py and are supposed to be replace with native implementations (rather than calling the tpm2 tools).

**Pull request closed without merging**